test/extensions/filters/http/ext_authz/ext_authz_test.cc

```diff
@@ -2008,6 +2008,55 @@ TEST_F(HttpFilterTest, EmitDynamicMetadata) {
   EXPECT_EQ(1U, config_->stats().ok_.value());
 }
 
+// Verify that when returning a Denied response with dynamic_metadata field set, the filter emits
+// dynamic metadata.
+TEST_F(HttpFilterTest, EmitDynamicMetadataWhenDenied) {
```

Comment:
Do we have a test that verifies that we don't emit it on errors?

Reply:
I'll add one. Thank you for reminding me.

Reply:
Added.